xblocks_contrib/word_cloud/word_cloud.py

```diff
@@ -0,0 +1,286 @@
+"""Word cloud is ungraded xblock used by students to
+generate and view word cloud.
+
+On the client side we show:
+If student does not yet answered - `num_inputs` numbers of text inputs.
+If student have answered - words he entered and cloud.
+"""
+import importlib.resources
+import uuid
+
+from web_fragments.fragment import Fragment
+from xblock.core import XBlock
+from xblock.fields import Boolean, Dict, Integer, List, Scope, String
+from xblock.utils.resources import ResourceLoader
+from xblock.utils.studio_editable import StudioEditableXBlockMixin
+
+resource_loader = ResourceLoader(__name__)
+
+
+# Make '_' a no-op so we can scrape strings. Using lambda instead of
+#  `django.utils.translation.ugettext_noop` because Django cannot be imported in this file
+def noop(text):
+    return text
+
+
+_ = noop
+
+
+def pretty_bool(value):
+    """Check value for possible `True` value.
+
+    Using this function we can manage different type of Boolean value
+    in xml files.
+    """
+    bool_dict = [True, "True", "true", "T", "t", "1"]
+    return value in bool_dict
+
+
+def resource_string(resource_path):
+    """Handy helper for getting resources from our kit."""
+    return importlib.resources.files(__package__).joinpath(resource_path.lstrip('/')).read_text()
+
+
+class WordCloudXBlock(StudioEditableXBlockMixin, XBlock):
```

Comment:
for consistency with edx-platform blocks, could you call it WordCloudBlock instead of WordCloudXBlock?

Reply:
Why should we prefer its consistency with edx-platform block rather than other XBlocks.
Is there any technical reason?

Reply:
My understanding is that most blocks use the `Block` suffix rather than `XBlock`.

There is no technical reason, it's purely stylistic.